arxiv/arxiv.py

```diff
@@ -626,32 +627,44 @@ def _format_url(self, search: Search, start: int, page_size: int) -> str:
         return self.query_url_format.format(urlencode(url_args))
 
     def _parse_feed(
-        self, url: str, first_page: bool = True
+        self,
+        url: str,
+        first_page: bool = True,
+        retries_left: int | None = None
```

Comment:
TODO: everything here-down should just take a `try_index`; check `try_index < self.num_retries`.